src/Bucket/Bucket.jl

```diff
@@ -0,0 +1,433 @@
+
+"""
+
+    struct BucketModel{
+         FT,
+         PS <: BucketModelParameters{FT},
+         ATM <: AbstractAtmosphericDrivers{FT},
+         RAD <: AbstractRadiativeDrivers{FT},
+         D,
+     } <: AbstractBucketModel{FT}
+
+Concrete type for the BucketModel, which store the model
+domain and parameters, as well as the necessary atmosphere
+and radiation fields for driving the model.
+$(DocStringExtensions.FIELDS)
+"""
+struct BucketModel{
+    FT,
+    PS <: BucketModelParameters{FT},
+    ATM <: AbstractAtmosphericDrivers{FT},
+    RAD <: AbstractRadiativeDrivers{FT},
+    D,
+} <: AbstractBucketModel{FT}
+    "Parameters required by the bucket model"
+    parameters::PS
+    "The atmospheric drivers: Prescribed or Coupled"
+    atmos::ATM
+    "The radiation drivers: Prescribed or Coupled"
+    radiation::RAD
+    "The domain of the model"
+    domain::D
+end
+
+function BucketModel(;
+    parameters::BucketModelParameters{FT, PSE},
+    domain::ClimaLSM.Domains.AbstractDomain,
+    atmosphere::ATM,
+    radiation::RAD,
+) where {FT, PSE, ATM, RAD}
+    args = (parameters, atmosphere, radiation, domain)
+    BucketModel{FT, typeof.(args)...}(args...)
+end
+
+prognostic_types(::BucketModel{FT}) where {FT} = (FT, FT, FT, FT)
+prognostic_vars(::BucketModel) = (:W, :T_sfc, :Ws, :S)
+auxiliary_types(::BucketModel{FT}) where {FT} = (FT, FT, FT, FT, FT)
+auxiliary_vars(::BucketModel) = (:q_sfc, :E, :LHF, :SHF, :R_n)
+
+"""
+    surface_fluxes( T_sfc::FT,
+                    q_sfc::FT,
+                    S::FT
+                    t::FT,
+                    parameters::P,
+                    atmos::PA,
+                    radiation::PR,
+                    ) where {FT <: AbstractFloat, P <: BucketModelParameters{FT},  PA <: PrescribedAtmosphere{FT}, PR <: PrescribedRadiation{FT}}
+
+Computes the surface flux terms at the ground for a standalone simulation:
+net radiation,  SHF,  LHF,
+as well as the water vapor flux (in units of m^3/m^2/s of water).
+Positive fluxes indicate flow from the ground to the atmosphere.
+
+It solves for these given atmospheric conditions, stored in `atmos`,
+ downwelling shortwave and longwave radiation (in `radiation`),
+model parameters, and the surface temperature and surface specific
+humidity.
+
+Currently, we only support soil covered surfaces.
+"""
+function surface_fluxes(
+    T_sfc::FT,
+    q_sfc::FT,
+    S::FT,
+    t::FT,
+    parameters::P,
+    atmos::PA,
+    radiation::PR,
+) where {
+    FT <: AbstractFloat,
+    P <: BucketModelParameters{FT},
+    PA <: PrescribedAtmosphere{FT},
+    PR <: PrescribedRadiation{FT},
+}
+    @unpack ρ_atmos, T_atmos, u_atmos, q_atmos, h_atmos, ρ_sfc = atmos
+    @unpack α_soil, α_snow, z_0m, z_0b, S_c, earth_param_set = parameters
+    @unpack LW_d, SW_d = radiation
+    _σ = Stefan()
+    _ρ_liq = ρ_cloud_liq(earth_param_set)
+
+    # call surface fluxes for E, SHF, LHF
+    ts_sfc = Thermodynamics.PhaseEquil_ρTq(earth_param_set, ρ_sfc, T_sfc, q_sfc)
```

Comment:
Should we add a warning that if `ρ_sfc = ρ_atmos`, this will break the flux calculation? Perhaps this better suited for `SurfaceFluxes.jl` though.